Cargo.toml

```diff
@@ -1,5 +1,6 @@
 [workspace]
 resolver = "2"
+package.rust-version = "1.80"
```

Comment:
I don't know about using a workspace-level MSRV... it's fine to require latest version for `stacks-node`, but we might want to be more conservative with stacks-common and stackslib, as those are used by other projects

Reply:
I thought about that too, but now when `clarity`, `stacks-common` and `stackslib` all use `LazyCell` and/or `LazyLock`, they need to be at 1.80 in any case. We can keep the `rust-version` flag local to the packages if that's preferred, I just figured it's easier to keep the whole project at an announced MSRV :shrug: No strong opinions here.

Reply:
While it's technically correct because the MSRV is 1.80 for all packages right now, and even though it can be overriden in package-level Cargo.toml files, this implies that we have some global MSRV policy which applies to all packages, which I don't think we want. So IMO it's better not to do this, but I don't really have strong feelings on it, and I'd approve either way

Reply:
@jbencin @obycode who need to weigh in here?  Should the ClarityWASM folks and Clarinet folks weigh in? cc @hugocaillard

Reply:
@jcnelson Hugo's on vacation right now, but I checked and Clarinet builds on Rust 1.80 without warnings. There are commit messages like this in the history:

```
445ea332 (4 weeks ago) Hugo C: fix: clippy warning v1.80 (#1518)
```

So this PR shouldn't be a problem for them

Reply:
Alright, it's fine by me as long as its easy to change back if someone complains.